modAionImpl/src/org/aion/zero/impl/blockchain/PendingTxCache.java

```diff
@@ -0,0 +1,160 @@
+/*******************************************************************************
+ * Copyright (c) 2017-2018 Aion foundation.
+ *
+ *     This file is part of the aion network project.
+ *
+ *     The aion network project is free software: you can redistribute it
+ *     and/or modify it under the terms of the GNU General Public License
+ *     as published by the Free Software Foundation, either version 3 of
+ *     the License, or any later version.
+ *
+ *     The aion network project is distributed in the hope that it will
+ *     be useful, but WITHOUT ANY WARRANTY; without even the implied
+ *     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *     See the GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with the aion network project source files.
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Aion foundation.
+ *
+ ******************************************************************************/
+
+package org.aion.zero.impl.blockchain;
+
+import org.aion.base.type.Address;
+import org.aion.log.AionLoggerFactory;
+import org.aion.log.LogEnum;
+import org.aion.zero.types.AionTransaction;
+import org.apache.commons.collections4.map.LRUMap;
+import org.slf4j.Logger;
+
+import java.math.BigInteger;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class PendingTxCache {
+
+    private Map<Address, Map<BigInteger,AionTransaction>> pendingTx;
+    protected static final Logger LOG = AionLoggerFactory.getLogger(LogEnum.GEN.name());
+
+    public PendingTxCache() {
+        pendingTx = Collections.synchronizedMap(new LRUMap<>(1_000_000));
```

Comment:
I feel like this value may be too big, heres my reasoning:

from https://ethereum.stackexchange.com/questions/1106/is-there-a-limit-for-transaction-size, at 3.1m gas the average block size is ~800kb, so if we apply the same logic to our limit, it means we can roughly be flooded with transactions that have a size of 2.4mb. This translates to a theoretical limit of 2.4 TB of memory use given an attacker incentivized to flood this cache.